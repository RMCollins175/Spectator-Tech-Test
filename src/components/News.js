import React, { Component } from "react";
import DiscoverNews from "./DiscoverNews";
import NavBar from "./NavBar";
import NewsList from "./NewsList.js";

import Axios from "axios";

const baseAPI =
  "http://newsapi.org/v2/top-headlines?country=gb&apiKey=55f245758a054a6381d493b655144574";

const searchAPI = "http://newsapi.org/v2/everything?q="
const apiKey = "&apiKey=55f245758a054a6381d493b655144574"

export default class News extends Component {
  constructor(props) {
    super(props);

    this.state = {
      results: []
    };
  }

//   async searchNews(keyword) {
//     let response = await Axios.get(`${searchAPI}${keyword}${apiKey}`)
//     console.log(response)
//   }

  async componentDidMount() {
    let response = await Axios.get(`${baseAPI}`);
    console.log(response.data.articles);
    this.setState({ results: response.data.articles });
  }

  render() {
    const { results } = this.state;

    return (
      <>
        <NavBar 
        searchNews={keyword => this.searchNews(keyword)} 
        />
        <DiscoverNews />
        <NewsList 
        newsStories={results || []} 
        />
      </>
    );
  }
}
