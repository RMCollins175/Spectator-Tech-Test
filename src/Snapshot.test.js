import React from "react";
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16";
import News from "./components/News";


Enzyme.configure({ adapter: new Adapter() })
const flushPromises = () => new Promise(resolve => setImmediate(resolve));

it('News snapshot test', async () => {
  const wrapper = shallow(<News />);
  await flushPromises();
  wrapper.update();
  expect(wrapper).toMatchSnapshot();
});