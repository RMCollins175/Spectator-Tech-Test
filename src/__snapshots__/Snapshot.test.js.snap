// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News snapshot test 1`] = `ShallowWrapper {}`;
